### ISSUE TITLE
auto connect not working 

### ISSUE THREAD

**Issue by hashoom8** [jneilliii/OctoPrint-BambuPrinter#22]
if the printer was powered off , and than turned on  , octoprint doesn't connect automatically
i must click on connect every time powering on the printer 

my old ender 5 it connect automatically

**Comment by jneilliii** [comment]
true, there's no reattempt at connecting once a connection is lost. with a physical connection to the pi on your ender 5 that can be detected easily by the OS. For this plugin it would require polling the ip address after a connection is lost to know when the printer comes back on. I personally never power my printer off so not sure when I would get around to adding this feature when you can just press the connect button.

**Comment by hashoom8** [comment]
> true, there's no reattempt at connecting once a connection is lost. with a physical connection to the pi on your ender 5 that can be detected easily by the OS. For this plugin it would require polling the ip address after a connection is lost to know when the printer comes back on. I personally never power my printer off so not sure when I would get around to adding this feature when you can just press the connect button.

not everyone using this plug in the same way everyone using it differently for me I am using a cloud connection and what I do is using AI failure detection so sometimes i start a print on the slicer and the print failed and become a huge miss because I forgot to press the connect button

**Comment by hashoom8** [comment]
or you can add an option in the settings to check the IP address every 5 minutes and if it's available reconnect automatically

**Comment by jneilliii** [comment]
> or you can add an option in the settings to check the IP address every 5 minutes and if it's available reconnect automatically

yeah, this was kind of how I would think to do it. but instead of making it an option would just automatically start polling the IP on disconnect say every minute. 

BTW, which device do you have?

**Comment by hashoom8** [comment]
> > or you can add an option in the settings to check the IP address every 5 minutes and if it's available reconnect automatically
> 
> yeah, this was kind of how I would think to do it. but instead of making it an option would just automatically start polling the IP on disconnect say every minute.
> 
> BTW, which device do you have?

raspberry pi 4 8gb and bambu lab A1 + pi cam V3

**Comment by hashoom8** [comment]
> > or you can add an option in the settings to check the IP address every 5 minutes and if it's available reconnect automatically
> 
> yeah, this was kind of how I would think to do it. but instead of making it an option would just automatically start polling the IP on disconnect say every minute.
> 
> BTW, which device do you have?

another problem after octoprint update to 1.10.1 ,  can't connect to the printer

**Comment by jneilliii** [comment]
There is no OctoPrint 1.10.1 and I've been using 1.10.0 for quite some time without issue. The A1 is very finicky about who is connected and who isn't. If you open the device tab of slicer and start getting data from the printer the printer will stop sending info to this plugin until a reconnect. That's unfortunately an issue with the printer itself and the same reconnect watchdog we were discussing is what would hopefully fix that too.

**Comment by hashoom8** [comment]
> There is no OctoPrint 1.10.1 and I've been using 1.10.0 for quite some time without issue. The A1 is very finicky about who is connected and who isn't. If you open the device tab of slicer and start getting data from the printer the printer will stop sending info to this plugin until a reconnect. That's unfortunately an issue with the printer itself and the same reconnect watchdog we were discussing is what would hopefully fix that too.

sorry 1.10.0 
![Screenshot 2024-04-27 234108](https://github.com/jneilliii/OctoPrint-BambuPrinter/assets/46507374/1cb416dc-8a8a-47de-b21d-ff7f8b1f3f60)

**Comment by jneilliii** [comment]
so it's dying on attempting to read the file list from the printer via ftps.

**Comment by hashoom8** [comment]
> so it's dying on attempting to read the file list from the printer via ftps.

yes , that happens after the update 1.10.0

**Comment by jneilliii** [comment]
Can you try enabling this checkbox in OctoPrint's settings, restarting OctoPrint and let me know if it makes any difference? 

![image](https://github.com/jneilliii/OctoPrint-BambuPrinter/assets/5249455/24c86020-1c86-4afc-ad53-99a43a10bf89)

**Comment by hashoom8** [comment]
> Can you try enabling this checkbox in OctoPrint's settings, restarting OctoPrint and let me know if it makes any difference?
> 
> ![image](https://private-user-images.githubusercontent.com/5249455/326222089-24c86020-1c86-4afc-ad53-99a43a10bf89.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQyNjczMTksIm5iZiI6MTcxNDI2NzAxOSwicGF0aCI6Ii81MjQ5NDU1LzMyNjIyMjA4OS0yNGM4NjAyMC0xYzg2LTRhZmMtYWQ1My05OWE0M2ExMGJmODkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjhUMDExNjU5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDUzMjgwMTY5ZWYxYzBmY2NhMGMxMTZmZmE4MjA0ZmRkMmNlODEyZmE4YWE5ZTgwOWFjNzEyODhmNGVkM2IyNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.I3uKETiHnSW6huA8otzC5RD6CC1FGGxyrHiNXHx3J9k)

it's checked by default
![Screenshot 2024-04-28 041904](https://github.com/jneilliii/OctoPrint-BambuPrinter/assets/46507374/03544f94-c6f7-47a0-9f02-6d9b68463fe3)

**Comment by jneilliii** [comment]
Ah, wasn't sure if I had unchecked it or not. In my current setup it was unchecked actually.

**Comment by hashoom8** [comment]
> Ah, wasn't sure if I had unchecked it or not. In my current setup it was unchecked actually.

i did try to uncheck it , and still no connect

**Comment by jneilliii** [comment]
enable debug logging in OctoPrint's logging section for all plugin.bambu_printer options and lets see if there are any hints in octoprint.log or the bambu logs that get generated.

**Comment by hashoom8** [comment]
> enable debug logging in OctoPrint's logging section for all plugin.bambu_printer options and lets see if there are any hints in octoprint.log or the bambu logs that get generated.

[serial.log](https://github.com/jneilliii/OctoPrint-BambuPrinter/files/15140823/serial.log)
[plugin_bambu_printer_serial.log](https://github.com/jneilliii/OctoPrint-BambuPrinter/files/15140824/plugin_bambu_printer_serial.log)

**Comment by hashoom8** [comment]
> enable debug logging in OctoPrint's logging section for all plugin.bambu_printer options and lets see if there are any hints in octoprint.log or the bambu logs that get generated.

https://drive.google.com/file/d/1gcZBUWKtI16zXkfxHTNNZGUvMDmoN7aJ/view?usp=sharing

**Comment by hashoom8** [comment]
> enable debug logging in OctoPrint's logging section for all plugin.bambu_printer options and lets see if there are any hints in octoprint.log or the bambu logs that get generated.

[octoprint (1).log](https://github.com/jneilliii/OctoPrint-BambuPrinter/files/15142234/octoprint.1.log)

**Comment by jneilliii** [comment]
You're using cloud connection to bambu online? 

> 2024-04-28 16:05:36,272 - octoprint.plugins.bambu_printer.BambuPrinter - DEBUG - connecting via local mqtt: False

and looking at some of the following errors, seems you are experiencing a network related issue with the underlying python module I'm using. 

> 2024-04-28 16:05:39,457 - pybambu - ERROR - A1: Exception. Type: <class 'OSError'> Args: [Errno 113] No route to host

**Comment by hashoom8** [comment]
> You're using cloud connection to bambu online?
> 
> > 2024-04-28 16:05:36,272 - octoprint.plugins.bambu_printer.BambuPrinter - DEBUG - connecting via local mqtt: False
> 
> and looking at some of the following errors, seems you are experiencing a network related issue with the underlying python module I'm using.
> 
> > 2024-04-28 16:05:39,457 - pybambu - ERROR - A1: Exception. Type: <class 'OSError'> Args: [Errno 113] No route to host

yes cloud connection  , so how can i fix it

**Comment by hashoom8** [comment]
> You're using cloud connection to bambu online?
> 
> > 2024-04-28 16:05:36,272 - octoprint.plugins.bambu_printer.BambuPrinter - DEBUG - connecting via local mqtt: False
> 
> and looking at some of the following errors, seems you are experiencing a network related issue with the underlying python module I'm using.
> 
> > 2024-04-28 16:05:39,457 - pybambu - ERROR - A1: Exception. Type: <class 'OSError'> Args: [Errno 113] No route to host

problem solved , i restart all my home network
